src/facade/dragonfly_connection.cc

```diff
@@ -369,14 +365,13 @@ void Connection::HandleRequests() {
     } else {
       cc_.reset(service_->CreateContext(peer, this));
       if (breaker_cb_) {
-        break_poll_id_ =
-            socket_->PollEvent(POLLERR | POLLHUP, [this](int32_t mask) { this->OnBreakCb(mask); });
+        socket_->RegisterOnErrorCb([this](int32_t mask) { this->OnBreakCb(mask); });
       }
 
       ConnectionFlow(peer);
 
-      if (break_poll_id_ != UINT32_MAX) {
-        socket_->CancelPoll(break_poll_id_);
+      if (breaker_cb_) {
```

Comment:
if you do this, you must make sure `CancelOnErrorCb()` can be called multiple times. we had some issues with shutting down a connection and data race around this codepath. you can see I used to reset break_poll_id_ to avoid calling CancelPoll if OnBreakCb was called first.

Reply:
Also, we must ensure that `OnBreakCb` won't be called once `cc_.reset();` is called. Currently, we have such cases.

Reply:
About the race with `cc_.reset();` - how does it work with io_uring?